### PR TITLE
`zen_get_categories_image`, ensure return is a string

### LINKED ISSUE #6843: function zen_get_categories_image can return null or string
Since the db default is null, if the category exists but no image has been defined, it returns null, not string.

https://github.com/zencart/zencart/blob/417f0347558a13768af625cba0596cfff9a7812e/includes/functions/functions_categories.php#L790-L805

### PR DESCRIPTION
**Pull request by lat9**
Fixes #6843

Note: Actual correction on line 807; remaining changes are reformatting/refactoring.

### CODE REVIEW

**Review by torvista — COMMENTED** [review]
Since there is an effort to handle
$result->EOF
I take that to mean the category id was invalid...should some debug notice be issued for that?

**Comment by lat9** [comment]
> Since there is an effort to handle $result->EOF I take that to mean the category id was invalid...should some debug notice be issued for that?

Probably, but that's outside of the scope of the PR and the associated issue.

File: includes/functions/functions_categories.php [hunk 1]
```diff
@@ -793,15 +793,18 @@ function zen_get_category_description($category_id, $language_id = null): string
...
+    $sql = "SELECT categories_image FROM " . TABLE_CATEGORIES . " WHERE categories_id = " . (int)$category_id;
+    $result = $db->Execute($sql, 1);
 
-    if ($result->EOF) return '';
+    if ($result->EOF) {
+        return '';
+    }
 
-    return $result->fields['categories_image'];
+    return (string)$result->fields['categories_image'];
```
Comment (drbyte) [inline includes/functions/functions_categories.php:807]: Given that the situation we're "preventing" is a `null` being returned, would it be more human-intuitive (for future looking at the code) to use null-coalesce here?

ie: 
```php
return $result->fields['categories_image'] ?? '';
```

(technically that could allow the deletion of the ->EOF check too, but I'm not opposed to keeping that)

Comment (lat9) [reply includes/functions/functions_categories.php:807]: I'd considered that, but felt that simply casting to a string was more 'obvious' and potentially more performant.

**Pull request merged**